src/py/flwr/cli/build.py

```diff
@@ -105,38 +114,63 @@ def build(
 
     toml_contents = tomli_w.dumps(conf)
 
-    with zipfile.ZipFile(fab_filename, "w", zipfile.ZIP_DEFLATED) as fab_file:
-        fab_file.writestr("pyproject.toml", toml_contents)
-
-        # Continue with adding other files
-        for root, _, files in os.walk(app, topdown=True):
-            files = [
-                f
-                for f in files
-                if not ignore_spec.match_file(Path(root) / f)
-                and f != fab_filename
-                and Path(f).suffix in allowed_extensions
-                and f != "pyproject.toml"  # Exclude the original pyproject.toml
-            ]
-
-            for file in files:
-                file_path = Path(root) / file
-                archive_path = file_path.relative_to(app)
-                fab_file.write(file_path, archive_path)
-
-                # Calculate file info
-                sha256_hash = get_sha256_hash(file_path)
-                file_size_bits = os.path.getsize(file_path) * 8  # size in bits
-                list_file_content += f"{archive_path},{sha256_hash},{file_size_bits}\n"
-
-        # Add CONTENT and CONTENT.jwt to the zip file
-        fab_file.writestr(".info/CONTENT", list_file_content)
+    with tempfile.NamedTemporaryFile(suffix=".zip", delete=False) as temp_file:
+        temp_filename = temp_file.name
+
+        with zipfile.ZipFile(temp_filename, "w", zipfile.ZIP_DEFLATED) as fab_file:
+            write_to_zip(fab_file, "pyproject.toml", toml_contents)
+
+            # Continue with adding other files
+            for root, _, files in os.walk(app, topdown=True):
```

Comment:
Could you use `for file_path in app.rglob("*"):` instead? For consistency and readability, it would be better to stick with pathlib throughout, though the original code mixes the two.